build/lib/src/builder/build_step.dart

```diff
@@ -26,19 +25,29 @@ abstract class BuildStep implements AssetReader, AssetWriter {
   @override
   Future<bool> hasInput(AssetId id);
 
+  /// Reads an [Asset] by [id] as a [List<int>].
+  @override
+  Future<List<int>> readAsBytes(AssetId id);
+
   /// Reads an [Asset] by [id] as a [String] using [encoding].
   @override
   Future<String> readAsString(AssetId id, {Encoding encoding: UTF8});
 
-  /// Outputs an [Asset] using the current [AssetWriter], and adds [asset] to
-  /// [outputs].
+  /// Outputs an [BytesAsset] using the current [AssetWriter].
+  ///
+  /// Most `Builder` implementations should not need to `await` this Future
```

Comment:
I suppose this bit of information could be worth adding. Feel free to keep the method signature and doc comment here